.buildkite/pipeline.yml

```
@@ -1,5 +1,38 @@
 # yaml-language-server: $schema=https://raw.githubusercontent.com/buildkite/pipeline-schema/main/schema.json
 
+env:
+  SETUP_GOLANG_VERSION: '1.17.6'
+
 steps:
-  - label: "Example test"
-    command: echo "Hello!"
+  - label: ":linux: Test on Go ${SETUP_GOLANG_VERSION}"
+    key: test
+    command:
+      - ".buildkite/scripts/test.sh"
+    agents:
+      image: golang:${SETUP_GOLANG_VERSION}
+      cpu: "8"
+      memory: "4G"
+    artifact_paths:
+      - "build/junit-*.xml"
+
+  - label: ":junit: Junit annotate"
+    plugins:
+      - junit-annotate#v2.4.1:
+          artifacts: "build/junit-*.xml"
+          fail-build-on-error: true
+    agents:
+      provider: "gcp" #junit plugin requires docker
+    depends_on:
+      - step: "test"
+        allow_failure: true
+
+  - label: ":linux: Microbench"
+    key: benchmark
+    command:
+      - ".buildkite/scripts/bench.sh"
```

Maybe we can keep the same shellscript under `.ci` to be used in .buildkite.
In this case, here it could be directly used that file:
```suggestion
      - ".ci/bench.sh"
```

WDYT ? @leo-ri @alexsapran